### PR TITLE
Refactor hmi capabilities test

### PR DESCRIPTION
**Pull request by ydementieiev**
This PR is **ready** for review.

### Summary
Refactor hmi_capabilities_test.

Some of changes:
- Deleted class variable (`file_cache_name_` and `file_name_`) and used constants instead.
- Proper `void SetUp()` setting.
- Using the class variables (`hmi_capabilities_`, `mock_app_mngr_`) instead of creating this same local inside each test.
- Delete unused variables, and variables which prepared in `void SetUp()`.
- Created `void CreateFile` for comfort create and check that file created.
- Fix extra calls to `LoadCapabilitiesFromFile`, for now we load only file which we need in current test.

If you have any refactoring suggestions please let me know,
and maybe during review you will have some idea for refactoring `TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile)` it's very big and I couldn't come up with anything good with it.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 1]
```diff
@@ -286,84 +283,78 @@ class HMICapabilitiesTest : public ::testing::Test {
...
+  }
+
+  void DeleteFileIfExists(const std::string& file_path) {
     if (file_system::FileExists(file_path)) {
       EXPECT_TRUE(::file_system::DeleteFile(file_path));
     }
   }
 
-  void SetCooperating();
+  std::shared_ptr<HMICapabilitiesForTesting> hmi_capabilities_;
```
Comment (AByzhynar) [inline src/components/application_manager/test/hmi_capabilities_test.cc:327]: @ydementieiev 
- why do you have header file `#include "application_manager/hmi_capabilities.h"` if you use HMICapabilitiesForTesting ? Do you use somewhere in the test real HMI Capabilities? 
- Why do we need at all this class `HMICapabilitiesForTesting`? Can't we test real HMICapabilities class?

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 2]
```diff
@@ -286,84 +283,78 @@ class HMICapabilitiesTest : public ::testing::Test {
...
 
   EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
               CommonLanguageFromString(_))
       .WillRepeatedly(Invoke(TestCommonLanguageFromString));
   EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
               CommonLightNameFromString(_))
       .WillRepeatedly(Invoke(TestCommonLightNameFromString));
-  ASSERT_TRUE(hmi_capabilities_test->LoadCapabilitiesFromFile());
+
+  hmi_capabilities_->Init(last_state_wrapper_);
```
Comment (AByzhynar) [inline src/components/application_manager/test/hmi_capabilities_test.cc:345]: @ydementieiev I would rename it to `hmi_capabilities_for_testing` as it is different from real HMI capabilities.

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 1 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: @AByzhynar I double checked,and deleted HMICapabilitiesForTesting, now we using real HMICapabilities class. Thank you

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 2 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc:347]: @AByzhynar  Now we using real HMICapabilities class.

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 3]
```diff
@@ -402,17 +405,24 @@ TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
   EXPECT_EQ(hmi_apis::Common_Language::FR_CA,
             static_cast<hmi_apis::Common_Language::eType>(
                 tts_supported_languages_so[2].asInt()));
+}
+
+TEST_F(HMICapabilitiesTest,
+       LoadCapabilitiesFromFile_CheckSpeechCapabilitiesSuccessLoadAndConvert) {
```
Comment (mvorobio) [inline src/components/application_manager/test/hmi_capabilities_test.cc:411]:
```suggestion
       LoadCapabilitiesFromFile_CheckSpeechCapabilities_SuccessLoadAndConvert) {
```

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 4]
```diff
@@ -286,84 +283,83 @@ class HMICapabilitiesTest : public ::testing::Test {
...
+            hmi_capabilities_->active_vr_language());
   EXPECT_EQ(hmi_apis::Common_Language::EN_US,
-            hmi_capabilities_test->active_tts_language());
+            hmi_capabilities_->active_tts_language());
+}
 
-  // Check UI languages
+TEST_F(HMICapabilitiesTest,
+       LoadCapabilitiesFromFile_CheckUILanguages_SuccessLoadAndConvert) {
+  SetUpLoadCapabilitiesFromFileExpectation();
```
Comment (mvorobio) [inline src/components/application_manager/test/hmi_capabilities_test.cc:359]: @ydementieiev do we need to use `SetUpLoadCapabilitiesFromFileExpectation` for all test cases?

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 3 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: @mvorobio Thank you, changed

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 4 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: @mvorobio no we don't need , I changed name to more [correct](https://github.com/LuxoftSDL/sdl_core/pull/17/commits/679f3d332f8eab9b627ab268df7009b457cde7b3#diff-c1d0a5558de3ab3e31ce8813484ebefeR326)

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 5]
```diff
@@ -493,8 +502,34 @@ TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
...
+}
+
+TEST_F(HMICapabilitiesTest,
+       LoadCapabilitiesFromFile_CheckTextFields_SuccessLoadAndConvert) {
+  SetUpLanguageAndLightCapabilitiesExpectation();
+  hmi_capabilities_->Init(last_state_wrapper_);
+  const auto display_capabilities_so =
+      *(hmi_capabilities_->display_capabilities());
+
+  const uint32_t text_len =
```
Comment (AByzhynar) [inline src/components/application_manager/test/hmi_capabilities_test.cc:514]: @ydementieiev Don't use conversion. Method length() returns `size_t`

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 6]
```diff
@@ -493,8 +502,34 @@ TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
...
+      display_capabilities_so[hmi_response::text_fields].length();
+  EXPECT_NE(0u, text_len);
+  for (uint32_t i = 0; i < text_len; ++i) {
```
Comment (AByzhynar) [inline src/components/application_manager/test/hmi_capabilities_test.cc:517]: @ydementieiev Use size_t

File: src/components/application_manager/test/hmi_capabilities_test.cc [hunk 7]
```diff
@@ -493,8 +502,34 @@ TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
...
+    const int32_t index = static_cast<int32_t>(i);
```
Comment (AByzhynar) [inline src/components/application_manager/test/hmi_capabilities_test.cc:518]: @ydementieiev Possible type overflow.

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 5 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: @AByzhynar change to size_t in this [PR](https://github.com/LuxoftSDL/sdl_core/pull/17/commits/53f85366b6a7cfc3fc2ccd60e33886690db075de)

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 6 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: @AByzhynar change to size_t in this [PR](https://github.com/LuxoftSDL/sdl_core/pull/17/commits/53f85366b6a7cfc3fc2ccd60e33886690db075de)

File: src/components/application_manager/test/hmi_capabilities_test.cc (hunk 7 above)
Comment (ydementieiev) [reply src/components/application_manager/test/hmi_capabilities_test.cc]: @AByzhynar removed this [in this PR](https://github.com/LuxoftSDL/sdl_core/pull/17/commits/53f85366b6a7cfc3fc2ccd60e33886690db075de)

**Pull request merged**